src/compiler/compiler.rs

```diff
@@ -1065,13 +1160,14 @@ mod test {
         let f = TestFixture::new();
         let creator = new_creator();
         let pool = CpuPool::new(1);
+        let cwd : PathBuf = "/tmp".into();
         next_command(
             &creator,
             Ok(MockChild::new(exit_status(0), "foo\nbar\ngcc", "")),
         );
-        let c = detect_compiler(&creator, &f.bins[0], &[], &pool, None)
+        let c = detect_compiler(creator.clone(), &f.bins[0], f.tempdir.path(), &[], &pool, None)
```

Comment:
Can't you just pass `creator` here and all the places following?

Reply:
Not sure I understand what you mean

Reply:
No, I mean calling `let c = detect_compiler(creator, ...)...`, since -- having not actually had the borrow checker check this -- I don't think `creator` is actually used after this point.

Reply:
Understood.